app/app_test.go

```diff
@@ -0,0 +1,54 @@
+package app
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tm-db"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/simapp"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+func TestExport(t *testing.T) {
+	db := db.NewMemDB()
+	app := NewInitApp(log.NewTMLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, 0)
+	setGenesis(app)
+
+	// Making a new app object with the db, so that initchain hasn't been called
+	newApp := NewInitApp(log.NewTMLogger(log.NewSyncWriter(os.Stdout)), db, nil, true, 0)
+	_, _, err := newApp.ExportAppStateAndValidators(false, []string{})
+	require.NoError(t, err, "ExportAppStateAndValidators should not have an error")
+}
+
+// ensure that black listed addresses are properly set in bank keeper
+func TestBlackListedAddrs(t *testing.T) {
```

Comment:
Hey mate, can you explain what the point of this blacklistedAddrs test is for? And what requirement are you trying to for-fill here?

Reply:
Tests were a direct lift from Gaia. Free stuff and all that. If we want to modify them/have our own, we should come up with a list of things we want to test for and I can write them out. In terms of what's there, TestExport makes sense to me, but I agree, we don't need TestBlackListedAddrs. Is there anything you want me to add in terms of unit testing? PR is for stacking/slashing but we can add other stuff too.

Reply:
I just kept this test , it doesnt fail 
We dont have blacklsited address as such ,but there is no harm in having a test case ?